develop/parachain-devs/parachain-maintenance/runtime-upgrades.md

```diff
@@ -0,0 +1,89 @@
+
+
+!!!info "Why 'Forkless'?"
+    Traditional blockchains require a hard fork when upgrading the state transition function of their chain. A hard fork requires all node operators to download the latest executable and typically restart their nodes. For distributed and decentralized blockchain production networks, coordination of hard fork upgrades can be a complex process.
+
+Because the runtime is part of the blockchain state, network maintainers can leverage the capabilities of the blockchain for trustless, decentralized consensus to securely make enhancements to the runtime.
+
+In FRAME, the [`system`](https://paritytech.github.io/polkadot-sdk/master/frame_system/index.html){target=\_blank} pallet defines the [`set_code`](https://paritytech.github.io/polkadot-sdk/master/frame_system/pallet/enum.Call.html#variant.set_code){target=\_blank} extrinsic that is used to update the Wasm code for the runtime. 
+
+!!!info "Parachain vs Solo Chain Upgrades"
+    While the `set_code` extrinsic is sufficient for updating the runtime in solo chain setups, parachains must undergo a different procedure of calling `authorize_upgrade`, followed by `apply_authorized_upgrade`. These extrinsics ensure that the relay chain is aware of the upgrade, and that it gets applied.
+
+When updating a runtime, there is a differentiation in adding functionality to the runtime versus updating existing functionality. In many cases, updating existing functionality, such as how a pallet stores state, would require a [storage migration](#storage-migrations).
+
+## Runtime Versioning
+
+The component that selects the runtime execution environment to communicate with is called the executor.  Although you can override the default execution strategies for custom scenarios, in most cases the executor selects the appropriate binary to use by evaluating and comparing key parameters from the native and Wasm runtime binaries.
+
+To provide this parameter information to the executor process, the runtime includes a [runtime version struct](https://paritytech.github.io/polkadot-sdk/master/sp_version/struct.RuntimeVersion.html){target=\_blank} similar to the following:
+
+```rust
+--8<-- 'code/develop/parachain-devs/parachain-maintenance/runtime_version.rs'
+```
+
+The parameters in the struct provide the following information to the executor:
+
+- **`spec_name`** - The identifier for the different runtimes
+- **`impl_name`** - The name of the implementation of the spec. This is of little consequence for the node and serves only to differentiate code of different implementation teams
+- **`authoring_version`** - The version of the authorship interface. An authoring node won't attempt to author blocks unless this is equal to its native runtime
+- **`spec_version`** - The version of the runtime specification
+
+    ??? info "Additional information"
```

Comment:
Ah yeah, not a huge fan of these dropdowns. It sort of distracts you from the actual definition haha

Reply:
Took them out ;-)